modules/transactions.js

```diff
@@ -818,23 +820,86 @@ Transactions.prototype.shared = {
 	 * @todo Add description of the function
 	 */
 	getTransactionsCount(cb) {
-		library.db.transactions.count().then(
-			transactionsCount =>
-				setImmediate(cb, null, {
-					confirmed: transactionsCount,
-					unconfirmed: Object.keys(__private.transactionPool.unconfirmed.index)
-						.length,
-					unprocessed: Object.keys(__private.transactionPool.queued.index)
-						.length,
-					unsigned: Object.keys(__private.transactionPool.multisignature.index)
-						.length,
-					total:
-						transactionsCount +
-						Object.keys(__private.transactionPool.unconfirmed.index).length +
-						Object.keys(__private.transactionPool.queued.index).length +
-						Object.keys(__private.transactionPool.multisignature.index).length,
-				}),
-			() => setImmediate(cb, 'Failed to count transactions')
+		async.waterfall(
+			[
+				function getTransactionCountFromCache(waterCb) {
+					modules.cache.getJsonForKey(
+						modules.cache.KEYS.transactionCount,
+						(err, data) => {
+							if (err) {
+								// If some issue in cache we will fallback to database
+								return setImmediate(waterCb, null, null);
+							}
+
+							return setImmediate(waterCb, null, data ? data.total : null);
+						}
+					);
+				},
+
+				function getTransactionCountFromDb(cachedCount, waterCb) {
+					if (cachedCount) {
+						return setImmediate(waterCb, null, cachedCount, null);
+					}
+
+					library.db.transactions
+						.count()
+						.then(transactionsCount =>
+							setImmediate(waterCb, null, cachedCount, transactionsCount)
+						);
+				},
+
+				function getTransactionCountToCache(cachedCount, dbCount, waterCb) {
```

Comment:
In the step above we use `transactionsCount`, here `dbCount`.  Should be nice to keep naming consistent.